network/notifee.go

```diff
@@ -0,0 +1,38 @@
+package network
+
+import (
+	lp2phost "github.com/libp2p/go-libp2p/core/host"
+	lp2pnetwork "github.com/libp2p/go-libp2p/core/network"
+	ma "github.com/multiformats/go-multiaddr"
+)
+
+type NotifeeService struct {
+	eventChannel chan<- Event
+}
+
+func newNotifeeService(host lp2phost.Host, eventChannel chan<- Event) *NotifeeService {
+	notifee := &NotifeeService{
+		eventChannel: eventChannel,
+	}
+	host.Network().Notify(notifee)
+	return notifee
+}
+
+func (n *NotifeeService) Connected(_ lp2pnetwork.Network, conn lp2pnetwork.Conn) {
+	n.eventChannel <- &ConnectEvent{PeerID: conn.RemotePeer()}
+}
+
+func (n *NotifeeService) Disconnected(_ lp2pnetwork.Network, conn lp2pnetwork.Conn) {
+	n.eventChannel <- &DisconnectEvent{PeerID: conn.RemotePeer()}
+}
+
+func (n *NotifeeService) Listen(_ lp2pnetwork.Network, _ ma.Multiaddr) {
+	// Handle listen event if needed.
+	// TODO: Log
```

Comment:
Add log here please. It is very useful in future.
Pass logger to NotifeeService